pkg/app/server/service/apiservice/service.proto

```diff
@@ -229,7 +229,7 @@ message RegisterEventRequest {
     string name = 1 [(validate.rules).string.min_len = 1];
     string data = 2 [(validate.rules).string.min_len = 1];
     map<string,string> labels = 3 [(validate.rules).map.keys.string.min_len = 1, (validate.rules).map.values.string.min_len = 1];
-    map<string,string> contexts = 4;
+    map<string,string> contexts = 4 [(validate.rules).map.keys.string.min_len = 1, (validate.rules).map.keys.string.pattern = "^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$", (validate.rules).map.values.string.min_len = 1];
```

Comment:
The regexp might be wrong.
The git implementations seem to allow double or more hyphens continuously like `test--hoge` or start with a hyphen like `-test`.
The pattern might be like `^[\-a-zA-Z0-9]+$`.

Reply:
@Warashi 
Yes, I know that.
I don't plan to do the same as validation for git.
It is because the trailer format is not so concrete.

So I wanted to express the format like a kebab case `test-hoge` in PipeCD.

Reply:
Okay, I got your point.